### ISSUE TITLE
continue extension failing onstart

### ISSUE THREAD

**Issue by Nanayeb34** [continuedev/continue#1233]
Type: <b>Bug</b>

 i installed Continue in wsl2. however, it doesn't work as expected. Instead i get the uncaught error below. no matter how many times i reload the extension, it still doesn't work but breaks with an `onStartupFinished` error below
 `The specified module could not be found. \?\c:\Users\Sam.vscode\extensions\continue.continue-0.9.124-win32-x64\bin\napi-v3\win32\x64\onnxruntime_binding.node`

<img width="642" alt="image" src="https://github.com/continuedev/continue/assets/69251896/f65874ee-119c-4a85-8d76-f5e30d6e99f0">

Extension version: 0.9.124
VS Code version: Code 1.89.0 (b58957e67ee1e712cebf466b995adf4c5307b2bd, 2024-05-01T02:09:22.859Z)
OS version: Windows_NT x64 10.0.22631
Modes:
Remote OS version: Linux x64 5.15.146.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|23.84GB (8.40GB free)|
|Process Argv|--crash-reporter-id e06d819a-bc66-47b0-810c-b9fd9649036f|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.15.146.1-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2111)|
|Memory (System)|11.60GB (7.67GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
ccp2r6:30993542
dsvsc020:30976470
pythonait:31006305
chatpanelt:31018789
dsvsc021:30996838
jg8ic977:31013176
pythoncenvptcf:31022791
dwnewjupytercf:31035177

```

</details>

**Comment by Nanayeb34** [comment]
@sestinj  could you kindly help here

**Comment by Nanayeb34** [comment]
yes @sestinj , the file exists at this location  `"C:\Users\Sam\.vscode\extensions\continue.continue-0.8.25-win32-x64\bin\napi-v3\win32\x64\onnxruntime_binding.node"` on my local machine

**Comment by sestinj** [comment]
Good to know, thank you! I will dig around onnxruntime_node to see what I can find

**Comment by AndreasRosenberger** [comment]
Hello,

I have the same issue on Windows VSC 1.89 and on Nixos VSC 1.84.

**Comment by sestinj** [comment]
@AndreasRosenberger there is ongoing conversation about NixOS [here](https://github.com/continuedev/continue/issues/821). There's no complete solution yet, but I believe in a few situations it's been discovered what dependencies are needed

**Comment by Atomesh** [comment]
Hi @sestinj 

I encountered a similar issue, with my error message being `The specified module could not be found. ?\c:\Users\AtomVW.vscode\extensions\continue.continue-0.8.27-win32-x64\bin\napi-v3\win32\x64\onnxruntime_binding.node`. This resembles the error Nanayeb34 reported: `The specified module could not be found. ?\c:\Users\Sam.vscode\extensions\continue.continue-0.9.124-win32-x64\bin\napi-v3\win32\x64\onnxruntime_binding.node`.

I noticed that the concatenation of the user directory and .vscode seems to be incorrect, suggesting a simple path error.
`Sam.vscode` `AtomVW.vscode`

I also observed that both Nanayeb34 and I are using Windows 10, so I attempted to install it on another machine running Windows 11, where I didn't encounter the same issue. Therefore, I speculate that this might be a path error specific to Windows 10?

**Comment by seros1521** [comment]
@sestinj, @AndreasRosenberger, @Nanayeb34, @Atomesh 

I encountered the same problem.
windows 10 
vscode 1.87.2
continue 0.8.27

Found this discussion:
https://stackoverflow.com/questions/69859120/the-specified-module-could-not-be-found-d-next-js-firstapp-node-modules-n

```
The issue is because the package is failing due to a missing library on the computer. fs-search requires Microsoft Visual C++ 2015 Redistributable (x64).

You can get the download here -> https://aka.ms/vs/17/release/vc_redist.x64.exe
```

After installing this everything worked.

**Comment by fry69** [comment]
These are very outdated versions. Current VSCode is 1.90.0 and current Continue extension is 0.8.40 (Continue is getting relevant changes, even with minor release, so I'd recommend to stay as much as possible up-to-date).

**Comment by RomneyDa** [comment]
+1 to outdated, thanks @fry69. Closing for now
